kaal-infrastructure/src/main/kotlin/cz/eman/kaal/infrastructure/api/retrofit/KaalRetrofitCaller.kt

```diff
@@ -140,4 +223,60 @@ interface KaalRetrofitCaller {
         message.logError(message, throwable)
         return Result.error(error = ErrorResult(code = code, message = message, throwable = throwable))
     }
+
+    /**
+     * Handles call exception. Any exception that is handled is used to create [Result.Error]. It makes sure the call
+     * does not crash the app and it receives information about what happened.
+     *
+     * @param ex Exception to handle
+     * @param errorMessage used to modify [Result.Error] message
+     * @return [Result] with type [T] - always creates [Result.Error]
+     * @see errorResult
+     * @since 0.9.0
+     */
+    private fun <T> handleCallException(ex: Exception, errorMessage: String?): Result<T> {
+        val errorCode = when (ex) {
+            is InvalidData -> return Result.error(error = ex.errorResult)
+            is SocketTimeoutException -> AdditionalErrorCode.SOCKET_TIMEOUT
+            is UnknownHostException -> AdditionalErrorCode.UNKNOWN_HOST
+            else -> ErrorCode.UNDEFINED
+        }
+
+        return errorResult(
+            code = errorCode,
+            message = errorMessage ?: ex.message ?: ex.toString(),
+            throwable = ex
+        )
+    }
+
+    /**
+     * Handles response and returns proper [Result]. If the response is successful and body is not null it will use
+     * [map] function to map [Dto] object to [T]. If it is null it will create [Result.Success] with null body. If the
+     * response is not successful it will create an [createApiErrorResult].
+     *
+     * @param response to be handled and parsed
+     * @param map used to map [Dto] object to [T]
+     * @return @return [Result] with type [T] or null
+     * @see createApiErrorResult
+     * @since 0.9.0
+     */
+    private suspend fun <Dto, T> handleResponse(response: Response<Dto>, map: suspend (Dto) -> T): Result<T?> {
+        return when {
+            response.isSuccessful -> {
+                val body = response.body()
+                if (body != null) {
```

Comment:
isn't better to use Elvis?
```
body?.let {
   Result.success(map(body))
} ?:
 Result.success(null)
```

or better
`Result.success(body?.let{ map(body) })`